src/types/componentWithProvider.ts

```diff
@@ -0,0 +1,4 @@
+export type ComponentWithProviderProps<T> = {
```

Comment:
just curious why we named it Provider? i can't really tell that it has open and close functions with the name 🤔

Reply:
because it's used with Provider, it won't be limited to open and close. You have a suggestion? 🤔

Reply:
you mean this would be used for combined one where we can add more apart from open and close in the future? I see it is currently used as being intersected with the Component's types which makes sense.

Reply:
yes, for component which are using providers